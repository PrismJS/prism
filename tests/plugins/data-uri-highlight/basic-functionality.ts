import { createTestSuite } from '../../helper/prism-dom-util';


describe('Data-URI Highlight', () => {
	const { it } = createTestSuite({
		languages: ['css', 'markup'],
		plugins: 'data-uri-highlight'
	});

	it('should set prefix', ({ util }) => {
		util.assert.highlight({
			language: 'css',
			code: String.raw`
div {
    border: 40px solid transparent;
    border-image: 33.334% url('data:image/svg+xml,<svg xmlns="http://www.w3.org/2000/svg" width="30" height="30"> \
                          <circle cx="5" cy="5" r="5" fill="%23ab4"/><circle cx="15" cy="5" r="5" fill="%23655"/> \
                          <circle cx="25" cy="5" r="5" fill="%23e07"/><circle cx="5" cy="15" r="5" fill="%23655"/> \
                          <circle cx="15" cy="15" r="5" fill="hsl(15, 25%, 75%)"/> \
                          <circle cx="25" cy="15" r="5" fill="%23655"/><circle cx="5" cy="25" r="5" fill="%23fb3"/> \
                          <circle cx="15" cy="25" r="5" fill="%23655"/><circle cx="25" cy="25" r="5" fill="%2358a"/></svg>');
    padding: 1em;
    max-width: 20em;
    font: 130%/1.6 Baskerville, Palatino, serif;
}
`,
		});
	});
});
