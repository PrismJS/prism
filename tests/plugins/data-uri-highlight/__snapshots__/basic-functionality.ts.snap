// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data-URI Highlight should set prefix 1`] = `
"<span class=\\"token selector\\">div</span>
<span class=\\"token punctuation\\">{</span>
<span class=\\"token property\\">border</span>
<span class=\\"token punctuation\\">:</span>
40px solid transparent
<span class=\\"token punctuation\\">;</span>
<span class=\\"token property\\">border-image</span>
<span class=\\"token punctuation\\">:</span>
33.334%
<span class=\\"token url\\">
	<span class=\\"token function\\">url</span>
	<span class=\\"token punctuation\\">(</span>
	<span class=\\"token string url\\">
		'
		<span class=\\"token data-uri\\">
			data:image/svg+xml,
			<span class=\\"token language-markup\\">
				<span class=\\"token tag\\">
					<span class=\\"token punctuation\\">&lt;</span>
					<span class=\\"token tag\\">svg</span>
					<span class=\\"token attr-name\\">xmlns</span>
					<span class=\\"token attr-value\\">
						<span class=\\"token punctuation attr-equals\\">=</span>
						<span class=\\"token punctuation\\">\\"</span>
						http://www.w3.org/2000/svg
						<span class=\\"token punctuation\\">\\"</span>
					</span>
					<span class=\\"token attr-name\\">width</span>
					<span class=\\"token attr-value\\">
						<span class=\\"token punctuation attr-equals\\">=</span>
						<span class=\\"token punctuation\\">\\"</span>
						30
						<span class=\\"token punctuation\\">\\"</span>
					</span>
					<span class=\\"token attr-name\\">height</span>
					<span class=\\"token attr-value\\">
						<span class=\\"token punctuation attr-equals\\">=</span>
						<span class=\\"token punctuation\\">\\"</span>
						30
						<span class=\\"token punctuation\\">\\"</span>
					</span>
					<span class=\\"token punctuation\\">></span>
				</span>
				\\\\
				<span class=\\"token tag\\">
					<span class=\\"token punctuation\\">&lt;</span>
					<span class=\\"token tag\\">circle</span>
					<span class=\\"token attr-name\\">cx</span>
					<span class=\\"token attr-value\\">
						<span class=\\"token punctuation attr-equals\\">=</span>
						<span class=\\"token punctuation\\">\\"</span>
						5
						<span class=\\"token punctuation\\">\\"</span>
					</span>
					<span class=\\"token attr-name\\">cy</span>
					<span class=\\"token attr-value\\">
						<span class=\\"token punctuation attr-equals\\">=</span>
						<span class=\\"token punctuation\\">\\"</span>
						5
						<span class=\\"token punctuation\\">\\"</span>
					</span>
					<span class=\\"token attr-name\\">r</span>
					<span class=\\"token attr-value\\">
						<span class=\\"token punctuation attr-equals\\">=</span>
						<span class=\\"token punctuation\\">\\"</span>
						5
						<span class=\\"token punctuation\\">\\"</span>
					</span>
					<span class=\\"token attr-name\\">fill</span>
					<span class=\\"token attr-value\\">
						<span class=\\"token punctuation attr-equals\\">=</span>
						<span class=\\"token punctuation\\">\\"</span>
						%23ab4
						<span class=\\"token punctuation\\">\\"</span>
					</span>
					<span class=\\"token punctuation\\">/></span>
				</span>
				<span class=\\"token tag\\">
					<span class=\\"token punctuation\\">&lt;</span>
					<span class=\\"token tag\\">circle</span>
					<span class=\\"token attr-name\\">cx</span>
					<span class=\\"token attr-value\\">
						<span class=\\"token punctuation attr-equals\\">=</span>
						<span class=\\"token punctuation\\">\\"</span>
						15
						<span class=\\"token punctuation\\">\\"</span>
					</span>
					<span class=\\"token attr-name\\">cy</span>
					<span class=\\"token attr-value\\">
						<span class=\\"token punctuation attr-equals\\">=</span>
						<span class=\\"token punctuation\\">\\"</span>
						5
						<span class=\\"token punctuation\\">\\"</span>
					</span>
					<span class=\\"token attr-name\\">r</span>
					<span class=\\"token attr-value\\">
						<span class=\\"token punctuation attr-equals\\">=</span>
						<span class=\\"token punctuation\\">\\"</span>
						5
						<span class=\\"token punctuation\\">\\"</span>
					</span>
					<span class=\\"token attr-name\\">fill</span>
					<span class=\\"token attr-value\\">
						<span class=\\"token punctuation attr-equals\\">=</span>
						<span class=\\"token punctuation\\">\\"</span>
						%23655
						<span class=\\"token punctuation\\">\\"</span>
					</span>
					<span class=\\"token punctuation\\">/></span>
				</span>
				\\\\
				<span class=\\"token tag\\">
					<span class=\\"token punctuation\\">&lt;</span>
					<span class=\\"token tag\\">circle</span>
					<span class=\\"token attr-name\\">cx</span>
					<span class=\\"token attr-value\\">
						<span class=\\"token punctuation attr-equals\\">=</span>
						<span class=\\"token punctuation\\">\\"</span>
						25
						<span class=\\"token punctuation\\">\\"</span>
					</span>
					<span class=\\"token attr-name\\">cy</span>
					<span class=\\"token attr-value\\">
						<span class=\\"token punctuation attr-equals\\">=</span>
						<span class=\\"token punctuation\\">\\"</span>
						5
						<span class=\\"token punctuation\\">\\"</span>
					</span>
					<span class=\\"token attr-name\\">r</span>
					<span class=\\"token attr-value\\">
						<span class=\\"token punctuation attr-equals\\">=</span>
						<span class=\\"token punctuation\\">\\"</span>
						5
						<span class=\\"token punctuation\\">\\"</span>
					</span>
					<span class=\\"token attr-name\\">fill</span>
					<span class=\\"token attr-value\\">
						<span class=\\"token punctuation attr-equals\\">=</span>
						<span class=\\"token punctuation\\">\\"</span>
						%23e07
						<span class=\\"token punctuation\\">\\"</span>
					</span>
					<span class=\\"token punctuation\\">/></span>
				</span>
				<span class=\\"token tag\\">
					<span class=\\"token punctuation\\">&lt;</span>
					<span class=\\"token tag\\">circle</span>
					<span class=\\"token attr-name\\">cx</span>
					<span class=\\"token attr-value\\">
						<span class=\\"token punctuation attr-equals\\">=</span>
						<span class=\\"token punctuation\\">\\"</span>
						5
						<span class=\\"token punctuation\\">\\"</span>
					</span>
					<span class=\\"token attr-name\\">cy</span>
					<span class=\\"token attr-value\\">
						<span class=\\"token punctuation attr-equals\\">=</span>
						<span class=\\"token punctuation\\">\\"</span>
						15
						<span class=\\"token punctuation\\">\\"</span>
					</span>
					<span class=\\"token attr-name\\">r</span>
					<span class=\\"token attr-value\\">
						<span class=\\"token punctuation attr-equals\\">=</span>
						<span class=\\"token punctuation\\">\\"</span>
						5
						<span class=\\"token punctuation\\">\\"</span>
					</span>
					<span class=\\"token attr-name\\">fill</span>
					<span class=\\"token attr-value\\">
						<span class=\\"token punctuation attr-equals\\">=</span>
						<span class=\\"token punctuation\\">\\"</span>
						%23655
						<span class=\\"token punctuation\\">\\"</span>
					</span>
					<span class=\\"token punctuation\\">/></span>
				</span>
				\\\\
				<span class=\\"token tag\\">
					<span class=\\"token punctuation\\">&lt;</span>
					<span class=\\"token tag\\">circle</span>
					<span class=\\"token attr-name\\">cx</span>
					<span class=\\"token attr-value\\">
						<span class=\\"token punctuation attr-equals\\">=</span>
						<span class=\\"token punctuation\\">\\"</span>
						15
						<span class=\\"token punctuation\\">\\"</span>
					</span>
					<span class=\\"token attr-name\\">cy</span>
					<span class=\\"token attr-value\\">
						<span class=\\"token punctuation attr-equals\\">=</span>
						<span class=\\"token punctuation\\">\\"</span>
						15
						<span class=\\"token punctuation\\">\\"</span>
					</span>
					<span class=\\"token attr-name\\">r</span>
					<span class=\\"token attr-value\\">
						<span class=\\"token punctuation attr-equals\\">=</span>
						<span class=\\"token punctuation\\">\\"</span>
						5
						<span class=\\"token punctuation\\">\\"</span>
					</span>
					<span class=\\"token attr-name\\">fill</span>
					<span class=\\"token attr-value\\">
						<span class=\\"token punctuation attr-equals\\">=</span>
						<span class=\\"token punctuation\\">\\"</span>
						hsl(15, 25%, 75%)
						<span class=\\"token punctuation\\">\\"</span>
					</span>
					<span class=\\"token punctuation\\">/></span>
				</span>
				\\\\
				<span class=\\"token tag\\">
					<span class=\\"token punctuation\\">&lt;</span>
					<span class=\\"token tag\\">circle</span>
					<span class=\\"token attr-name\\">cx</span>
					<span class=\\"token attr-value\\">
						<span class=\\"token punctuation attr-equals\\">=</span>
						<span class=\\"token punctuation\\">\\"</span>
						25
						<span class=\\"token punctuation\\">\\"</span>
					</span>
					<span class=\\"token attr-name\\">cy</span>
					<span class=\\"token attr-value\\">
						<span class=\\"token punctuation attr-equals\\">=</span>
						<span class=\\"token punctuation\\">\\"</span>
						15
						<span class=\\"token punctuation\\">\\"</span>
					</span>
					<span class=\\"token attr-name\\">r</span>
					<span class=\\"token attr-value\\">
						<span class=\\"token punctuation attr-equals\\">=</span>
						<span class=\\"token punctuation\\">\\"</span>
						5
						<span class=\\"token punctuation\\">\\"</span>
					</span>
					<span class=\\"token attr-name\\">fill</span>
					<span class=\\"token attr-value\\">
						<span class=\\"token punctuation attr-equals\\">=</span>
						<span class=\\"token punctuation\\">\\"</span>
						%23655
						<span class=\\"token punctuation\\">\\"</span>
					</span>
					<span class=\\"token punctuation\\">/></span>
				</span>
				<span class=\\"token tag\\">
					<span class=\\"token punctuation\\">&lt;</span>
					<span class=\\"token tag\\">circle</span>
					<span class=\\"token attr-name\\">cx</span>
					<span class=\\"token attr-value\\">
						<span class=\\"token punctuation attr-equals\\">=</span>
						<span class=\\"token punctuation\\">\\"</span>
						5
						<span class=\\"token punctuation\\">\\"</span>
					</span>
					<span class=\\"token attr-name\\">cy</span>
					<span class=\\"token attr-value\\">
						<span class=\\"token punctuation attr-equals\\">=</span>
						<span class=\\"token punctuation\\">\\"</span>
						25
						<span class=\\"token punctuation\\">\\"</span>
					</span>
					<span class=\\"token attr-name\\">r</span>
					<span class=\\"token attr-value\\">
						<span class=\\"token punctuation attr-equals\\">=</span>
						<span class=\\"token punctuation\\">\\"</span>
						5
						<span class=\\"token punctuation\\">\\"</span>
					</span>
					<span class=\\"token attr-name\\">fill</span>
					<span class=\\"token attr-value\\">
						<span class=\\"token punctuation attr-equals\\">=</span>
						<span class=\\"token punctuation\\">\\"</span>
						%23fb3
						<span class=\\"token punctuation\\">\\"</span>
					</span>
					<span class=\\"token punctuation\\">/></span>
				</span>
				\\\\
				<span class=\\"token tag\\">
					<span class=\\"token punctuation\\">&lt;</span>
					<span class=\\"token tag\\">circle</span>
					<span class=\\"token attr-name\\">cx</span>
					<span class=\\"token attr-value\\">
						<span class=\\"token punctuation attr-equals\\">=</span>
						<span class=\\"token punctuation\\">\\"</span>
						15
						<span class=\\"token punctuation\\">\\"</span>
					</span>
					<span class=\\"token attr-name\\">cy</span>
					<span class=\\"token attr-value\\">
						<span class=\\"token punctuation attr-equals\\">=</span>
						<span class=\\"token punctuation\\">\\"</span>
						25
						<span class=\\"token punctuation\\">\\"</span>
					</span>
					<span class=\\"token attr-name\\">r</span>
					<span class=\\"token attr-value\\">
						<span class=\\"token punctuation attr-equals\\">=</span>
						<span class=\\"token punctuation\\">\\"</span>
						5
						<span class=\\"token punctuation\\">\\"</span>
					</span>
					<span class=\\"token attr-name\\">fill</span>
					<span class=\\"token attr-value\\">
						<span class=\\"token punctuation attr-equals\\">=</span>
						<span class=\\"token punctuation\\">\\"</span>
						%23655
						<span class=\\"token punctuation\\">\\"</span>
					</span>
					<span class=\\"token punctuation\\">/></span>
				</span>
				<span class=\\"token tag\\">
					<span class=\\"token punctuation\\">&lt;</span>
					<span class=\\"token tag\\">circle</span>
					<span class=\\"token attr-name\\">cx</span>
					<span class=\\"token attr-value\\">
						<span class=\\"token punctuation attr-equals\\">=</span>
						<span class=\\"token punctuation\\">\\"</span>
						25
						<span class=\\"token punctuation\\">\\"</span>
					</span>
					<span class=\\"token attr-name\\">cy</span>
					<span class=\\"token attr-value\\">
						<span class=\\"token punctuation attr-equals\\">=</span>
						<span class=\\"token punctuation\\">\\"</span>
						25
						<span class=\\"token punctuation\\">\\"</span>
					</span>
					<span class=\\"token attr-name\\">r</span>
					<span class=\\"token attr-value\\">
						<span class=\\"token punctuation attr-equals\\">=</span>
						<span class=\\"token punctuation\\">\\"</span>
						5
						<span class=\\"token punctuation\\">\\"</span>
					</span>
					<span class=\\"token attr-name\\">fill</span>
					<span class=\\"token attr-value\\">
						<span class=\\"token punctuation attr-equals\\">=</span>
						<span class=\\"token punctuation\\">\\"</span>
						%2358a
						<span class=\\"token punctuation\\">\\"</span>
					</span>
					<span class=\\"token punctuation\\">/></span>
				</span>
				<span class=\\"token tag\\">
					<span class=\\"token punctuation\\">&lt;/</span>
					<span class=\\"token tag\\">svg</span>
					<span class=\\"token punctuation\\">></span>
				</span>
			</span>
		</span>
		'
	</span>
	<span class=\\"token punctuation\\">)</span>
</span>
<span class=\\"token punctuation\\">;</span>
<span class=\\"token property\\">padding</span>
<span class=\\"token punctuation\\">:</span>
1em
<span class=\\"token punctuation\\">;</span>
<span class=\\"token property\\">max-width</span>
<span class=\\"token punctuation\\">:</span>
20em
<span class=\\"token punctuation\\">;</span>
<span class=\\"token property\\">font</span>
<span class=\\"token punctuation\\">:</span>
130%/1.6 Baskerville
<span class=\\"token punctuation\\">,</span>
Palatino
<span class=\\"token punctuation\\">,</span>
serif
<span class=\\"token punctuation\\">;</span>
<span class=\\"token punctuation\\">}</span>
"
`;
